app/models/archivematica_metadata.rb

```diff
@@ -0,0 +1,270 @@
+
+  def degree_fields
+    @thesis.degrees.each do |degree|
+      @csv_hash[:headers] << 'dc.description.degree'
+      @csv_hash[:headers] << 'thesis.degree.name'
+      @csv_hash[:headers] << 'mit.thesis.degree'
+
+      @thesis.files.each_with_index do |file, i|
+        degree_abbreviation(file, i, degree)
+        degree_name_dspace(file, i, degree)
+        degree_degree_type(file, i, degree)
+      end
+    end
+  end
+
+  def degree_abbreviation(file, file_number, degree)
+    @csv_hash["f#{file_number}".to_sym] << thesis_pdf_checker(file, degree.abbreviation)
+  end
+
+  def degree_name_dspace(file, file_number, degree)
+    @csv_hash["f#{file_number}".to_sym] << thesis_pdf_checker(file, degree.name_dspace)
+  end
+
+  def degree_degree_type(file, file_number, degree)
+    @csv_hash["f#{file_number}".to_sym] << thesis_pdf_checker(file, degree.degree_type&.name.to_s)
+  end
+
+  def non_repeating
+    @csv_hash[:headers] << ['filename', 'Level_of_DPCommitment', 'dc.title', 'dc.date.issued', 'dc.date.submitted',
+                            'dc.type', 'dc.description.abstract', 'dc.identifier.uri', 'BitstreamDescription',
+                            'BitstreamChecksumValue']
+    @thesis.files.each_with_index do |f, i|
+      @csv_hash["f#{i}".to_sym] << "data/#{f.blob.filename}"
+      @csv_hash["f#{i}".to_sym] << level_of_commitment(f)
+      @csv_hash["f#{i}".to_sym] << dc_title(f)
+      @csv_hash["f#{i}".to_sym] << dc_date_issued(f)
+      @csv_hash["f#{i}".to_sym] << dc_date_submitted(f)
+      @csv_hash["f#{i}".to_sym] << dc_type(f)
+      @csv_hash["f#{i}".to_sym] << dc_description_abstract(f)
+      @csv_hash["f#{i}".to_sym] << dc_identifier_uri(f)
+      @csv_hash["f#{i}".to_sym] << bitstream_description(f)
+      @csv_hash["f#{i}".to_sym] << bitstream_checksum_value(f)
+    end
+  end
+
+  def level_of_commitment(file)
+    thesis_pdf_checker(file, 'Level 3')
+  end
+
+  def dc_title(file)
+    thesis_pdf_checker(file, @thesis.title)
+  end
+
+  def dc_date_issued(file)
+    thesis_pdf_checker(file, @thesis.grad_date.strftime('%Y-%m'))
+  end
+
+  def dc_date_submitted(file)
+    thesis_pdf_checker(file, file.blob.created_at)
+  end
+
+  def dc_description_abstract(file)
+    thesis_pdf_checker(file, @thesis.abstract)
+  end
+
+  def dc_identifier_uri(file)
+    thesis_pdf_checker(file, @thesis.dspace_handle.to_s)
+  end
+
+  def bitstream_description(file)
+    # NOTE: these file purposes are repeated from sqs_message.rb and should be moved to a common location
+    file_purposes = { 'thesis_pdf' => 'Thesis PDF', 'thesis_source' => 'Thesis source', 'thesis_supplementary_file' =>
+                      'Supplementary file', 'proquest_form' => 'Proquest form', 'signature_page' => 'Signature page' }
+    file_purposes[file.purpose]
+  end
+
+  def bitstream_checksum_value(file)
+    base64_to_hex(file.checksum)
+  end
+
+  # Duplicated logic from dspace_publication_results_job.rb and should be considered form shared location
```

Comment:
This is now in the Checksums module: https://github.com/MITLibraries/thing/blob/main/app/lib/checksums.rb